project/profiles/capacity4more/modules/c4m/og/c4m_og/plugins/entityreference/selection/C4MOgSelectionHandler.class.php

```diff
@@ -90,6 +90,13 @@ public function buildEntityFieldQuery(
     $user_groups = og_get_groups_by_user(NULL, $group_type);
     $user_groups = $user_groups ? $user_groups : array();
 
+    $group_obj = og_context();
+    $node_type = $this->instance['bundle'];
+    if (!og_user_access($group_type, $group_obj['gid'], "create $node_type content")) {
+      // User does not have permission, falsify the query.
+      $query->propertyCondition($entity_info['entity keys']['id'], -1, '=');
```

Comment:
@ybaras 
so please write it down on the code.
these kind of numbers should not live in the code without a good explanation.
I would create a constant for that on the beginning of the code and replace it all.